tests/lib/rules/classic-decorator-no-classic-methods.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/classic-decorator-no-classic-methods');
+const RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: 'babel-eslint',
+  parserOptions: { ecmaVersion: 6, sourceType: 'module' },
+});
+ruleTester.run('classic-decorator-no-classic-methods', rule, {
+  valid: [
+    `
+      const Foo = EmberObject.extend({
+        foo() {
+          this.get('bar');
+        }
+      })
+
+      class Foo extends Bar {
+        foo() {
+          get(this, 'bar');
+          this.baz();
+        }
+      }
+
+      @classic
+      class Foo extends Bar {
+        foo() {
+          this.get('bar');
+        }
+      }
+
+      class Foo {
+        foo() {
+          this.get('bar');
+        }
+      }
+    `,
+  ],
+
+  invalid: [
+    {
+      code: `
+        class Foo extends Bar {
+          foo() {
+            this.get('bar');
+          }
+        }
+      `,
+      errors: [
+        {
+          message: /The this.get\(\) method is a classic ember object method, and can't be used in octane classes./,
+        },
```

Comment:
You should also check `type` in each failing test case to make sure the error is being flagged on the right type of element.